shpc/main/modules/__init__.py

```diff
@@ -359,3 +363,45 @@ def install(self, name, tag=None, **kwargs):
             name = "%s:%s" %(name, tag.name)
         logger.info("Module %s was created." % name)
         return container_path
+
+    # Module software can choose how to handle each of these cases
+    def _no_default_version(self, version_file):
+        return
+
+    def _sys_module_default_version(self, version_file):
+        return
+
+    def _set_default_version(self, version_file, tag):
+        """
+        Set the default version to the given tag
+        """
+        template = self._load_template("default_version")
+        utils.write_file(version_file, template.render(version=tag))
+
+    def write_version_file(self, uri):
+        """
+        Create the .version file, if there is a template for it.
+        """
+        version_dir = os.path.join(self.settings.module_base, uri)
+        version_file = os.path.join(version_dir, ".version")
+
+        # No default versions
```

Comment:
The reason we want to include tag is:

1. to be explicit
2. so all the version setting functions have a consistent set of arguments
3. we cannot be sure how a module system (or future module system) might want to use the tag. It just happens to be the case that we don't need it for specific cases now, but we cannot guarantee this forever.

So I think my preference would be to restore the tag variable, and that way we can theoretically be very explicit about the version (tag) we are providing. I should arguably be able to write my own little wrapper to do some weird custom logic given a specific tag.